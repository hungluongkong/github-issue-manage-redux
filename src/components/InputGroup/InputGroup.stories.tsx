import React from 'react';
import { storiesOf } from '@storybook/react';
import InputGroup from '.';
import { InputType } from '../../types';
import { PreviewArea, Code, DetailTable } from '../commonStyled';

// Detail of props
const propsDetail: JSX.Element = DetailTable({
  data: [
    { name: 'label', type: 'string' },
    { name: 'defaultValue', type: 'string' },
    { name: 'placeholder', type: 'string' },
    { name: 'isRequire', type: 'boolean', default: 'false' },
    { name: 'textRef', type: 'React.RefObject<HTMLInputElement | HTMLTextAreaElement>' },
  ],
});

storiesOf('Input Group', module)
  .add('Input Group Types', () => (
    <>
      <h3>Input Group Types</h3>
      <p>Input and TextArea</p>
      <PreviewArea>
        <InputGroup
          label="Title"
          defaultValue="default text"
          placeholder="Your input here"
          inputType={InputType.Input}
          isRequire
        />
        <InputGroup
          label="Title"
          defaultValue="default text"
          placeholder="Your input here"
          inputType={InputType.TextArea}
        />
      </PreviewArea>
      <Code>
        {`
          <InputGroup
            label="Title"
            defaultValue="default text"
            placeholder="Your input here"
            inputType={InputType.Input}
            isRequire={true}
          />
          <InputGroup
            label="Title"
            defaultValue="default text"
            placeholder="Your input here"
            inputType={InputType.TextArea}
          />
        `}
      </Code>
      {propsDetail}
    </>
  ))
  .add('Input Group Label', () => (
    <>
      <h3>Input Group Label</h3>
      <p>Input with/without Label</p>
      <PreviewArea>
        <InputGroup
          defaultValue="Without Label"
          placeholder="Your input here"
          inputType={InputType.Input}
        />
        <InputGroup
          label="Title"
          defaultValue="With Label"
          placeholder="Your input here"
          inputType={InputType.Input}
        />
      </PreviewArea>
      <Code>
        {`
          <InputGroup
            defaultValue="Without Label"
            placeholder="Your input here"
            inputType={InputType.Input}
          />
          <InputGroup
            label="Title"
            defaultValue="With Label"
            placeholder="Your input here"
            inputType={InputType.Input}
          />
        `}
      </Code>
      {propsDetail}
    </>
  ));
