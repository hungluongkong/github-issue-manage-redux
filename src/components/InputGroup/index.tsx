import React from 'react';
import styled from 'styled-components';
import TextArea from '../TextArea';
import Input from '../Input';
import { InputGroupProps, InputType } from '../../types';

const Label = styled.label`
  font-weight: 600;
  display: inline-block;
  padding: 0 15px;
  width: 100px;
`;

const InputGroupWrapper = styled.div`
  display: flex;
  width: 100%;
  align-items: center;
`;

const InputWrapper = styled.div<{ isTextArea: boolean }>`
  min-height: ${(props) => (props.isTextArea ? 200 : 0)}px;
  flex-grow: 1;
  line-height: 20px;
  box-sizing: border-box;
`;

const InputGroup = ({
  defaultValue,
  label = '',
  inputType = InputType.Input,
  placeholder = '',
  isRequire = false,
  textRef = React.createRef<HTMLTextAreaElement | HTMLInputElement>(),
}: InputGroupProps): JSX.Element => {
  const InputElement = inputType === InputType.Input ? Input : TextArea;

  return (
    <InputGroupWrapper>
      {label && <Label>{label}</Label>}
      <InputWrapper isTextArea={inputType === InputType.TextArea}>
        <InputElement
          textRef={textRef}
          defaultValue={defaultValue}
          placeholder={placeholder}
          isRequire={isRequire}
        />
      </InputWrapper>
    </InputGroupWrapper>
  );
};

export default InputGroup;
