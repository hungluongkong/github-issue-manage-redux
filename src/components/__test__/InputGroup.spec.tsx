import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import InputGroup from '../InputGroup';

describe('Unit test for Input component', () => {
  it('snapshot', () => {
    const input = create(<InputGroup defaultValue="text" />);
    expect(input).toMatchSnapshot();
  });

  it('should render label', () => {
    const label = 'Title';
    const input = mount(<InputGroup defaultValue="text" label={label} />);
    expect(input.text()).toEqual(label);
  });

  it('should not render label', () => {
    const input = mount(<InputGroup defaultValue="text" />);
    expect(input.text()).toEqual('');
  });
});
