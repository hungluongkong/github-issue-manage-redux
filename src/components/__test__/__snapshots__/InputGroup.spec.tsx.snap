// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for Input component snapshot 1`] = `
<div
  className="sc-AxhCb dtChlF"
>
  
  <div
    className="sc-AxhUy eURWlK"
  >
    <input
      className="sc-AxirZ cBNEtA"
      defaultValue="text"
      onChange={[Function]}
      placeholder=""
      required={false}
    />
  </div>
</div>
`;
